src/backend/api/Fusion.Resources.Api/Controllers/Contracts/ContractsController.cs

```diff
@@ -0,0 +1,189 @@
+        {
+
+            return Created($"/projects/{projectIdentifier}/contracts/{request.ContractNumber}", new ApiContract
+            {
+                Id = Guid.NewGuid(),
+                ContractNumber = request.ContractNumber,
+                Name = request.Name,
+                Description = request.Description,
+                StartDate = request.StartDate,
+                EndDate = request.EndDate,
+                Company = request.Company != null ? new ApiCompany { Id = Guid.NewGuid(), Identifier = request.Company.Identifier, Name = new Faker().Company.CompanyName() } : null,
+                CompanyRepPositionId = request.CompanyRepPositionId,
+                ContractResponsiblePositionId = request.ContractResponsiblePositionId,
+                ExternalCompanyRepPositionId = request.ExternalCompanyRepPositionId,
+                ExternalContractResponsiblePositionId = request.ExternalContractResponsiblePositionId
+            });
+        }
+
+        [HttpPut("/projects/{projectIdentifier}/contracts/{contractIdentifier}")]
+        public async Task<ActionResult<ApiContract>> UpdateProjectContract(string projectIdentifier, string contractIdentifier, [FromBody] ContractRequest request)
+        {
+
+            return Created($"/projects/{projectIdentifier}/contracts/{request.ContractNumber}", new ApiContract
+            {
+                Id = request.Id.GetValueOrDefault(Guid.NewGuid()),
+                ContractNumber = request.ContractNumber,
+                Name = request.Name,
+                Description = request.Description,
+                StartDate = request.StartDate,
+                EndDate = request.EndDate,
+                Company = request.Company != null ? new ApiCompany { Id = Guid.NewGuid(), Identifier = request.Company.Identifier, Name = new Faker().Company.CompanyName() } : null,
+                CompanyRepPositionId = request.CompanyRepPositionId,
+                ContractResponsiblePositionId = request.ContractResponsiblePositionId,
+                ExternalCompanyRepPositionId = request.ExternalCompanyRepPositionId,
+                ExternalContractResponsiblePositionId = request.ExternalContractResponsiblePositionId
+            });
+        }
+
+        [HttpPost("/projects/{projectIdentifier}/contracts/{contractIdentifier}/external-company-representative")]
+        public async Task<ActionResult> CreateContractExternalCompanyRep(string projectIdentifier, string contractIdentifier, [FromBody] ContractPositionRequest request)
+        {
+
+            return Ok();
+        }
+
+        [HttpPost("/projects/{projectIdentifier}/contracts/{contractIdentifier}/external-contract-responsible")]
+        public async Task<ActionResult> CreateContractExternalContractResp(string projectIdentifier, string contractIdentifier, [FromBody] ContractPositionRequest request)
+        {
+
+            return Ok();
+        }
+    }
+
+    public class ContractPositionRequest
```

Comment:
Move all these models to seperate files?

Reply:
yeh, will do when progressing a bit.. suspects a some changes are gonne come :)